### PR TITLE
[prompts] Correct PromptObject.initial's type

### PR DESCRIPTION
**Pull request by clmystes**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/terkelg/prompts#initial

### CODE REVIEW

File: types/prompts/index.d.ts [hunk 1]
```diff
@@ -78,7 +78,7 @@ declare namespace prompts {
         type: PromptType | Falsy | PrevCaller<T, PromptType | Falsy>;
         name: ValueOrFunc<T>;
         message?: ValueOrFunc<string>;
-        initial?: string | number | boolean | Date;
+        initial?: ValueOrFunc<string>;
```
Comment (Whoaa512) [inline types/prompts/index.d.ts:81]: [Numbers](https://github.com/terkelg/prompts#numbermessage-initial-max-min-style), [booleans](https://github.com/terkelg/prompts#togglemessage-initial-active-inactive), and [dates](https://github.com/terkelg/prompts#datemessage-initial-warn) are also valid values here as noted in the docs of other prompts functions

**Review by Whoaa512 — CHANGES_REQUESTED** [review]

File: types/prompts/index.d.ts (hunk 1 above)
Comment (clmystes) [reply types/prompts/index.d.ts]: Fixed.

**Review by Whoaa512 — APPROVED** [review]
Thank you for the changes and thorough tests! 💯

**Comment by clmystes** [comment]
Ready to merge

**Pull request merged**